### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by benlapETS**
Completed the GUI with pyqt:

- Added pyqt .ui design files
- Added pyqt class implementation files
- Adapted i18n translation

### CODE REVIEW

File: hanggame/__init__.py [hunk 1]
```diff
@@ -24,4 +24,34 @@
...
+        The login of The Hangman Game
+    **login.ui**:
+        The Qt ui design for the login dialog of The Hangman Game
+    **progress_bar.py**:
+        The progress bar of The Hangman Game
+    **progress_bar.ui**:
+        The Qt ui design for the progress bar of The Hangman Game
+    **scoreboard.py**:
+        The scoreboard of The Hangman Game
+    **scoreboard.ui**:
```
Comment (labesoft) [inline hanggame/__init__.py:51]: I would remove the .ui there as it is part of qt designer generated code

File: hanggame/game.py [hunk 2]
```diff
@@ -16,31 +16,25 @@
...
+        An empty string constants
+    **YES_NO_LIST**
         lists of yes/no options that the player may use to answer the play again
         question.
 """
 import logging
 import random
 
-from hanggame.greeter import Greeter, OUT_MSG_CONGRATS, OUT_MSG_LOSER, OUT_MSG_WINNER, OUT_MSG_COMPLAINTS, _
+from hanggame.greeter import Greeter
```
Comment (labesoft) [inline hanggame/game.py:30]: greeter module and Greeter class should be rename to console.Console

File: hanggame/__init__.py [hunk 3]
```diff
@@ -24,4 +24,34 @@
...
+        The Qt ui design for the scoreboard of The Hangman Game
+    **word_view.py**:
+        The word label view of The Hangman Game
+    **word_view.ui**:
+        The Qt ui design for the word label view of The Hangman Game
```
Comment (labesoft) [inline hanggame/__init__.py:56]: I would remove the .ui there as it is part of qt designer generated code

File: hanggame/game.py [hunk 4]
```diff
@@ -16,31 +16,25 @@
...
+from hanggame import i18n
```
Comment (labesoft) [inline hanggame/game.py:31]: i18n module is a good idea

File: hanggame/hangman.py [hunk 5]
```diff
@@ -27,24 +27,24 @@
...
-    "\t__|__"
+    "    _____",
+    "   |       |",
+    "   |       |",
+    "   |       |",
+    "   |",
+    "   |",
+    "   |",
+    "__|__"
 ]
```
Comment (labesoft) [inline hanggame/hangman.py:38]: These changes won't be enoughfor the GUI display

File: hanggame/i18n.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+-----------------------------------
+
+About this module
+-----------------
+The objective of this module is to offer The Hangman Game a centralized way of
+translating string throughout the whole application.
+
+File structure
+--------------
+*import*
```
Comment (labesoft) [inline hanggame/i18n.py:11]: imports are not documented here ?

File: hanggame/locales/en_CA/LC_MESSAGES/hanggame.po [hunk 7]
```diff
@@ -1,195 +1,248 @@
-# Translations template for PROJECT.
```
Comment (labesoft) [inline hanggame/locales/en_CA/LC_MESSAGES/hanggame.po:1]: Where is the procedure to translate the app ? Should be documented in the readme (developper part)

File: hanggame/ui/game_window.py [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+using Python. It uses innovative python libraries such as PyQt5 which helped
+to visually create the UI representation of the game and import directly into
+our code. That way we created highly modular, modifiable and testable UI code.
+
+File structure
+--------------
+*import*
+
+*constant*
+    None
```
Comment (labesoft) [inline hanggame/ui/game_window.py:16]: Undocumented imports and unnecessary constant None doc

File: hanggame/ui/login.py [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+The objective of this module is to create a dialog before The Hangman Game
+starts that offers the player to enter his name.
+
+File structure
+--------------
+*import*
+    **PyQt5: QtCore, QtGui, QtWidgets, uic**
+        Useful modules for a PyQt module
+
+*constant*
```
Comment (labesoft) [inline hanggame/ui/login.py:15]: useless

File: hanggame/ui/progress_bar.py [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+as PyQt5 which helps to visually create the UI representation of the progress
+during the game.
+
+File structure
+--------------
+*import*
+    **PyQt5: QtCore, QtGui, QtWidgets, uic**
+        Useful modules for a PyQt module
+
+*constant*
```
Comment (labesoft) [inline hanggame/ui/progress_bar.py:17]: useless

File: hanggame/ui/scoreboard.py [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+which helps to visually create the UI representation of the live score during
+the game.
+
+File structure
+--------------
+*import*
+    **PyQt5: QtCore, QtGui, QtWidgets, uic**
+        Useful modules for a PyQt module
+
+*constant*
```
Comment (labesoft) [inline hanggame/ui/scoreboard.py:17]: useless

File: hanggame/ui/word_view.py [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+could also display critical messages to the player.
+
+File structure
+--------------
+*import*
+    **PyQt5: QtCore, QtGui, QtWidgets, uic**
+        Useful modules for a PyQt module
+
+*constant*
+    None
```
Comment (labesoft) [inline hanggame/ui/word_view.py:18]: useless

File: hanggame/word.py [hunk 13]
```diff
@@ -85,6 +85,7 @@ def unmasked_set(self):
 
     def choose(self):
         """Chooses a new word from the word bank and assign it"""
+        self.show()
```
Comment (labesoft) [inline hanggame/word.py:88]: showing the word doesn't make sense here

File: tests/test_game.py [hunk 14]
```diff
@@ -10,6 +10,7 @@
 from unittest.mock import patch, MagicMock, call
 
 import hanggame
+import hanggame.i18n
```
Comment (labesoft) [inline tests/test_game.py:13]: should import from here: from hanggame import i18n

File: tests/test_greeter.py [hunk 15]
```diff
@@ -9,6 +9,8 @@
 from unittest.mock import MagicMock, call, patch
 
 from hanggame.greeter import *
+from hanggame.i18n import IN_MSG_LETTER, IN_MSG_REPLAY, OUT_MSG_ANSWER, OUT_MSG_GOODBYE, OUT_MSG_INVALID, OUT_MSG_LUCK, \
+    OUT_MSG_NB_ATTEMPT, OUT_MSG_READY, OUT_MSG_THANKS, OUT_MSG_WELCOME
```
Comment (labesoft) [inline tests/test_greeter.py:13]: from hanggame import i18n

**Review by ghost — APPROVED** [review]
Good job... left some comments though

**Pull request merged**